client/asset/btc/btc.go

```diff
@@ -401,9 +408,25 @@ func (btc *ExchangeWallet) Balance() (*asset.Balance, error) {
 	}, err
 }
 
-// Fund selects utxos (as asset.Coin) for use in an order. Any Coins returned
-// will be locked. Part of the asset.Wallet interface.
-func (btc *ExchangeWallet) Fund(value uint64, nfo *dex.Asset) (asset.Coins, error) {
+// FeeRate returns the current optimal fee rate in sat / byte.
+func (btc *ExchangeWallet) FeeRate() (uint64, error) {
+	feeResult, err := btc.node.EstimateSmartFee(1, &btcjson.EstimateModeConservative)
+	if err != nil {
+		return 0, err
+	}
+	if len(feeResult.Errors) > 0 {
+		return 0, fmt.Errorf(strings.Join(feeResult.Errors, "; "))
+	}
+	if feeResult.FeeRate == nil {
+		return 0, fmt.Errorf("no fee rate available")
+	}
+	satPerKB, err := btcutil.NewAmount(*feeResult.FeeRate)
+	return uint64(satPerKB) / 1000, err
```

Comment:
In @davecgh`s  words:
>It's almost never good practice to return non-zero values in the case errors, unless the value is specifically known to be good in that case (e.g. a reader that returns the number of bytes read until the error happened, etc).

Reply:
That's good practice, and we are actually doing that because we know `NewAmount` does it (and that `0 / s == 0`).